recipes-support/boost/boost/riscv32__NR_futex.patch

```diff
@@ -11,3 +19,16 @@
  // Some Android NDKs (Google NDK and older Crystax.NET NDK versions) don't define SYS_futex
  #if defined(SYS_futex)
  #define BOOST_LOG_SYS_FUTEX SYS_futex
+--- a/boost/fiber/detail/futex.hpp
++++ b/boost/fiber/detail/futex.hpp
+@@ -7,6 +7,10 @@
+ #ifndef BOOST_FIBERS_DETAIL_FUTEX_H
+ #define BOOST_FIBERS_DETAIL_FUTEX_H
+ 
++#if !defined(__NR_futex) && defined(__riscv) && __riscv_xlen == 32
```

Comment:
We don't need the `__riscv` checks here, this isn't RISC-V specific, just 64-bit time_t on 32-bit specific.

Reply:
other 32 bit architectures define `__NR_futex` so this is not needed for others.

Reply:
Other 32-bit architectures using 64-bit time_t don't. This includes Arc, but will be other architectures in the future.

Reply:
right, I meant old 32bit architectures, so what is the right condition for detecting this case ? perhaps 
`#if !defined(__NR_futex) && defined(__NR_futex_time64)` might get the right condition

Reply:
Yep, that seems better